src/main/java/dev/openfeature/javasdk/OpenFeatureClient.java

```diff
@@ -94,9 +95,14 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
             if (details == null) {
                 details = FlagEvaluationDetails.<T>builder().build();
             }
+            if (e instanceof OpenFeatureError) {
+                details.setErrorCode(((OpenFeatureError)e).getErrorCode());
+            } else {
+                details.setErrorCode(ErrorCode.GENERAL);
+            }
+            details.setMessage(e.getMessage());
             details.setValue(defaultValue);
-            details.setReason(Reason.ERROR);
-            details.setErrorCode(e.getMessage());
+            details.setReason(Reason.ERROR.toString());
```

Comment:
This is the only behavioral change - if we can get an errorCode, us it. Otherwise use `GENERAL`.

Also, get the message from the message. Previously, the code was coming from the message.